src/game/Mails/MailHandler.cpp

```diff
@@ -824,13 +824,27 @@ void WorldSession::HandleQueryNextMailTime(WorldPacket& /**recv_data*/)
             if (now < m->deliver_time)
                 continue;
 
-            data << ObjectGuid(HIGHGUID_PLAYER, m->sender); // sender guid
+            HighGuid guidType = HIGHGUID_PLAYER;
+
+            switch (m->messageType)
+            {
+                case MAIL_CREATURE:
+                    guidType = HIGHGUID_UNIT;
+                    break;
+                case MAIL_GAMEOBJECT:
+                    guidType = HIGHGUID_GAMEOBJECT;
+                    break;
+            }
+
+            data << ObjectGuid(guidType, m->sender); // sender guid
 
             switch (m->messageType)
             {
                 case MAIL_AUCTION:
-                    data << uint32(m->sender);              // auction house id
-                    data << uint32(MAIL_AUCTION);           // message type
+                case MAIL_CREATURE:
+                case MAIL_GAMEOBJECT:
+                    data << uint32(m->sender);              // sender's id
+                    data << uint32(m->messageType);         // message type
                     break;
                 default:
                     data << uint32(0);
```

Comment:
from my naive reading of the Mail class the `m->sender` and `m->messageType` values should always be defined and shouldn't require an additional switch
I think just adding them like
```cpp
data << static_cast<uint32>(m->sender);
data << static_cast<uint32>(m->messageType);
```
should be fine

Reply:
I tried to keep the original behavior for the other mail cases, so it would send 0 and 0 when mail is not from auction/character/gameobject. I will test to see if it would break anything to always send data, thanks for the idea!

Reply:
Tested with all cases, everything works exactly as before so I think it's safe to change.